documents/api/viewsets.py

```diff
@@ -59,14 +59,12 @@ def get_queryset(self):
         qs_filters = {}
 
         # Filter the Documents only for the user's Service
-        if service := self.request.service:
+        if service := get_service_from_request(self.request):
             qs_filters["document__service"] = service
 
         # If the user doesn't have permissions to view that Service,
         # only show the Documents that belong to them
-        if not user.has_perm(
-            ServicePermissions.VIEW_ATTACHMENTS.value, qs_filters.get("service")
-        ):
+        if not user.has_perm(ServicePermissions.VIEW_ATTACHMENTS.value, service):
```

Comment:
Can the service be accessed here? IIRC the scope of `service := get_service_from_request` should only be within the `if` clause. 🤔

Reply:
The assignment operator does not create a new scope i.e. the value is available outside the if clause.

Reply:
But since calling `get_service_from_request` will actually result in an exception if the service is not found, I'll remove the `if` clause.

Reply:
Ahhh good to know! I was under the impression that it only worked within the scope it had been declared 👌